assimilation_code/modules/observations/DEFAULT_obs_kind_mod.F90

```diff
@@ -608,81 +75,96 @@ module obs_kind_mod
 character(len=*), parameter :: revision = ''
 character(len=*), parameter :: revdate  = ''
 
-logical, save :: module_initialized = .false.
+! integer, parameters: max_defined_types_of_obs, max_defined_quantities 
+! generated and inserted by preprocess.
 
-!! max_defined_types_of_obs is really max_obs_types, and is generated by preprocess.
-
-!! PRIVATE ONLY TO THIS MODULE.  max_defined_types_of_obs is really max_obs_types,
-!! but since the original variable name is public it can't change without
-!! being non-backwards compatible.  for now, punt on both type and kind
-!! and start trying to use specific and generic (ok, which do map to type
-!! and kind, respectively).  using intermediate names might make the transition
-!! less painful.  right now, many(most) of the subroutine names or args
-!! which are public are using 'kind' where it needs to be 'type'.
-integer, parameter :: max_obs_specific = max_defined_types_of_obs
+logical, save :: module_initialized = .false.
 
 character(len=512) :: msg_string, msg_string1
 
-integer :: num_kind_assimilate, num_kind_evaluate
-integer :: num_kind_use_precomputed_FOs
+! @todo FIXME: what is the right len for a units and
+! other name/value pair strings?
+integer, parameter :: namelen  = 256
+integer, parameter :: valuelen = 256
 
-! Map from values of kind in obs_def to the fixed values in the list above.
+integer :: num_types_assimilate, num_types_evaluate 
+integer :: num_types_use_precomputed_FOs
+
+! @todo FIXME would this be clearer as 2 named integer arrays?
```

Comment:
```suggestion
```

Reply:
yes, this comment should be removed because i made the single 2d array into a derived type.